src/App.tsx

```diff
@@ -17,18 +53,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                applyQuery={applyQuery}
+                isTodoCompleted={isTodoCompleted}
+                setIsTodoComlpeted={setIsTodoComlpeted}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+
+              <TodoList
+                todos={visibleTodos}
+                setSelectedTodo={setSelectedTodo}
+                selectedTodo={selectedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo !== null
+        && (
+          <TodoModal
+            selectedTodo={selectedTodo}
+            setSelectedTodo={setSelectedTodo}
+          />
+        )}
```

Comment:
When you use some variable in a condition it will check if it exists and check for falseness

```suggestion
      {selectedTodo && (
          <TodoModal
            selectedTodo={selectedTodo}
            setSelectedTodo={setSelectedTodo}
          />
        )}
```

Reply:
👌fixed